application/src/main/java/net/corda/v5/application/persistence/PagedQuery.java

```diff
@@ -49,5 +49,33 @@
      */
     @Suspendable
     @NotNull
-    List<R> execute();
+    ResultSet<R> execute();
+
+    /**
+     * A representation of a result set returned by a paged query when executed.
+     *
+     * @param <T> Type of the results contained in the result set.
+     */
+    interface ResultSet<T> {
+
+        /**
+         * @return The results contained in the result set.
+         */
+        List<T> getResults();
+
+        /**
+         * @return The size of the result set.
+         */
+        Integer getSize();
+
+        /**
+         * @return The new offset from where the result set continues or 0 if the result set is finished.
+         */
+        Integer getNewOffset();
+
+        /**
+         * @return Whether the result set has a next page or not.
+         */
+        Boolean hasNextPage();
```

Comment:
I'm really not sure about this. When you run a DB query, for example, you don't know if there is a "next page", until you run the next query.
Having this on the API would mean you always have to run 2 DB queries to find out. That would be inefficient, and also be different to otherwise well understood APIs.

It's also confusing to talk about pages when we don't in the main api. There it was a conscious decision to use limits and offsets, not pages, as that again aligns with the JPA API.